kyo-scheduler/js/src/main/scala/kyo/scheduler/InternalClock.scala

```diff
@@ -0,0 +1,13 @@
+package kyo.scheduler
+
+import java.util.concurrent.Executor
+import scala.annotation.nowarn
+
+@nowarn
+final class InternalClock(executor: Executor = null):
+
+    def currentMillis(): Long = System.currentTimeMillis()
```

Comment:
Should it be nano time?

Reply:
`System.nanoTime` isn't monotonic across threads. It also helps to hide the imprecision of the clock.